### PR TITLE
Use secure socket layer for the metrics expose

### PR DESCRIPTION
**Pull request by SchSeba**
This commit add SSL for the metrics exposed by the linuxptp daemons

Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

**Comment by SchSeba** [comment]
```
hack/run-functests.sh
/home/sscheink/Documents/GolangProjects/bin/ginkgo
Running Suite: PTP e2e integration tests
========================================
Random Seed: 1592997310
Will run 9 of 9 specs

•••
------------------------------
• [SLOW TEST:166.019 seconds]
[ptp]
/home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:32
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:88
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:126
      The interfaces support ptp can be discovered correctly
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:142
------------------------------
• [SLOW TEST:75.922 seconds]
[ptp]
/home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:32
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:88
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:126
      The virtual interfaces should be not discovered by ptp
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:155
------------------------------
••
------------------------------
• [SLOW TEST:35.842 seconds]
[ptp]
/home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:32
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:88
    PTP Interfaces discovery
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:126
      Can provide a profile with higher priority
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:219
------------------------------
• [SLOW TEST:11.692 seconds]
[ptp]
/home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:32
  PTP e2e tests
  /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:88
    PTP metric is present
    /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:261
      on slave
      /home/sscheink/Documents/GolangProjects/src/github.com/openshift/ptp-operator/test/ptp/ptp.go:277
------------------------------

JUnit report was created: /tmp/artifacts/unit_report.xml

Ran 9 of 9 Specs in 309.388 seconds
SUCCESS! -- 9 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
```

File: deploy/clusterrole.yaml [hunk 1]
```diff
@@ -54,3 +54,9 @@ rules:
 - apiGroups: ["ptp.openshift.io"]
   resources: ["*"]
   verbs: ["*"]
+- apiGroups: ["authentication.k8s.io"]
+  resources: ["tokenreviews"]
+  verbs: ["create"]
+- apiGroups: ["authorization.k8s.io"]
+  resources: ["subjectaccessreviews"]
+  verbs: ["create"]
```
Comment (fedepaol) [inline deploy/clusterrole.yaml:62]: New line

File: test/utils/pods/pods.go [hunk 2]
```diff
@@ -3,146 +3,19 @@ package pods
...
-		return false, nil
-	})
-}
+	testclient "github.com/openshift/ptp-operator/test/utils/client"
+)
 
 // GetLog connects to a pod and fetches log
 func GetLog(p *corev1.Pod) (string, error) {
-	req := testclient.Client.Pods(p.Namespace).GetLogs(p.Name, &corev1.PodLogOptions{})
+	req := testclient.Client.Pods(p.Namespace).GetLogs(p.Name, &corev1.PodLogOptions{Container: "linuxptp-daemon-container"})
```
Comment (fedepaol) [inline test/utils/pods/pods.go:18]: Can you make this generic and pass the container name as a parameter? Otherwise it's' GetLogFromDaemonContainer

**Comment by fedepaol** [comment]
A couple of nits. I am also wondering if it would be better to have the test refactoring as a separate commit.

**Comment by SchSeba** [comment]
Done can you take another look please

**Pull request merged**